src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -11890,7 +11890,8 @@ void Unit::UpdateSpeed(UnitMoveType mtype)
             else             // Use not mount (shapeshift for example) auras (should stack)
                 main_speed_mod  = GetTotalAuraModifier(SPELL_AURA_MOD_INCREASE_FLIGHT_SPEED) + GetTotalAuraModifier(SPELL_AURA_MOD_INCREASE_VEHICLE_FLIGHT_SPEED);
 
-            non_stack_bonus += GetMaxPositiveAuraModifier(SPELL_AURA_MOD_FLIGHT_SPEED_NOT_STACK) / 100.0f;
+            // SPELL_AURA_MOD_MOUNTED_SPEED_NOT_STACK should increase flying mount speed as well.
+            non_stack_bonus += std::max(GetMaxPositiveAuraModifier(SPELL_AURA_MOD_FLIGHT_SPEED_NOT_STACK), GetMaxPositiveAuraModifier(SPELL_AURA_MOD_MOUNTED_SPEED_NOT_STACK)) / 100.0f;
```

Comment:
You're right, there are two existing serverside spells in spell_dbc that do just that. Thanks, didn't even think about checking there! :)